lib/runner/TestRunner.js

```diff
@@ -17,31 +17,40 @@ module.exports = class TestRunner {
     }
 
     async run(testFile, context) {
-        return new Promise((resolve, reject) => {
-            fs.readFile(testFile, "utf8", (error, data) => {
-                if (error) {
-                    reject(error);
-                }
-
-                try {
-                    const testSuite = this.parseContents(testFile, data);
-                    resolve(this.runSuite(testSuite, context));
-                } catch (e) {
-                    reject(e);
-                }
+        let file;
+        try {
+            file = fs.readFileSync(testFile, "utf8");
+        } catch (error) {
+            throw error;
+        }
 
-            });
-        });
+        // This may have already been called, but needs to be called again inside of the Jest runner
+        await Configuration.configure(this._config);
+        try {
+            const testSuite = this.parseContents(testFile, file);
+            const locales = testSuite.locales;
+            if (locales) {
+                const localesList = locales.split(",");
+                let results = await Promise.all(localesList.map(locale => {
+                    const testSuiteWithLocale = Object.assign(Object.create(testSuite), testSuite)
+                    testSuiteWithLocale.currentLocale = locale;
+                    return this.runSuite(testSuiteWithLocale, context);
```

Comment:
A new method should be added, runSuiteForLocale that takes a suite and a specific locale for which it should be executed. The bulk of the logic from the existing runSuite will go there. And runSuite will handle just iterating through the multiple locales (if there are many locales).

That way we can have both a runSuite and runFile entry point, both of which will execute for several locales.